kv_test.go

```diff
@@ -251,3 +251,51 @@ func TestKeyValueCreate(t *testing.T) {
 		t.Fatalf("Unexpected error code, got: %v", kerr.APIError().ErrorCode)
 	}
 }
+
+func TestKeyValueSourcing(t *testing.T) {
+	s := RunBasicJetStreamServer()
+	defer shutdownJSServerAndRemoveStorage(t, s)
+
+	nc, js := jsClient(t, s)
+	defer nc.Close()
+
+	kvA, err := js.CreateKeyValue(&KeyValueConfig{Bucket: "A"})
+	if err != nil {
+		t.Fatalf("Error creating kv: %v", err)
+	}
+
+	_, err = kvA.Create("keyA", []byte("1"))
+	if err != nil {
+		t.Fatalf("Error creating key: %v", err)
+	}
+
+	if _, err := kvA.Get("keyA"); err != nil {
+		t.Fatalf("Got error getting keyA from A: %v", err)
+	}
+
+	kvB, err := js.CreateKeyValue(&KeyValueConfig{Bucket: "B"})
+	if err != nil {
+		t.Fatalf("Error creating kv: %v", err)
+	}
+
+	_, err = kvB.Create("keyB", []byte("1"))
+	if err != nil {
+		t.Fatalf("Error creating key: %v", err)
+	}
+
+	kvC, err := js.CreateKeyValue(&KeyValueConfig{Bucket: "C", Sources: []*StreamSource{{Name: "A"}, {Name: "B"}}})
+	if err != nil {
+		t.Fatalf("Error creating kv: %v", err)
+	}
+
+	// Wait half a second to make sure it has time to populate the stream from it's sources
+	time.Sleep(100 * time.Millisecond)
```

Comment:
This looks flaky... why not just wrap below code with the `kvC.Get` with some retries until they work?

Reply:
It's not flaky, just a question of timing. Without it, the test passes like 99% of the time but I did see it fail once. IMHO it's expected as there is a small delay between the stream creation call returning and the stream being populated from it's sources, so I added the sleep to give the stream a little bit of time to come up and get populated with the two messages from the 2 sources and haven't seen it fail since. I actually lowered it further to 20 ms and it still passes every time.

Reply:
I did implement something like you suggest of checking the number of messages and waiting until it reaches 2 and then doing the gets - see the last commit.
And what happens is that the first size check always returns 0 so it does the sleep of 20ms then the next time around the stream is populated and it goes on with the gets, so the tests with this extra check before sleep always runs the same amount of time as with just the sleep.